--- avrdude.conf.in	2013-09-18 07:59:06.000000000 +0200
+++ avrdude.conf.in	2014-02-27 17:14:40.686503613 +0100
@@ -751,6 +751,15 @@
 ;
 
 programmer
+  id    = "arduinoisp";
+  desc  = " ";
+  type  = "usbtiny";
+  connection_type = usb;
+  usbvid     = 0x2341;
+  usbpid     = 0x0049;
+;
+
+programmer
   id    = "butterfly";
   desc  = "Atmel Butterfly Development Board";
   type  = "butterfly";
