--- avrdude.conf.in	2015-04-02 23:54:05.862556159 +0000
+++ avrdude.conf.in	2015-04-03 00:07:02.156406668 +0000
@@ -9234,7 +9234,7 @@
      avr910_devcode   = 0x20;
      signature        = 0x1e 0x93 0x0b;
      reset            = io;
-     chip_erase_delay = 4500;
+     chip_erase_delay = 400000;
 
      pgm_enable       = "1 0 1 0  1 1 0 0    0 1 0 1  0 0 1 1",
                         "x x x x  x x x x    x x x x  x x x x";
@@ -9300,7 +9300,7 @@
 			  "  x   x   x   x      x   x   x   x";
 
 	mode		= 0x41;
-	delay		= 6;
+	delay		= 12;
 	blocksize	= 4;
 	readsize	= 256;
        ;
@@ -9309,8 +9309,8 @@
          size            = 8192;
          page_size       = 64;
          num_pages       = 128;
-         min_write_delay = 4500;
-         max_write_delay = 4500;
+         min_write_delay = 30000;
+         max_write_delay = 30000;
          readback_p1     = 0xff;
          readback_p2     = 0xff;
          read_lo         = "  0   0   1   0    0   0   0   0",
