Committed
Binutils 2.19
--------------------------------------------------------------------------------
--- gas/config/tc-avr.c.old	2007-09-26 09:40:21.793625000 -0600
+++ gas/config/tc-avr.c	2007-09-26 09:41:19.887375000 -0600
@@ -98,6 +98,7 @@ static struct mcu_type_s mcu_types[] =
   {"attiny25",   AVR_ISA_TINY2,   bfd_mach_avr2},
   {"attiny45",   AVR_ISA_TINY2,   bfd_mach_avr2},
   {"attiny85",   AVR_ISA_TINY2,   bfd_mach_avr2},
+  {"attiny43u",  AVR_ISA_TINY2,   bfd_mach_avr2},
   {"atmega603",  AVR_ISA_M603,    bfd_mach_avr3}, /* XXX -> m103 */
   {"atmega103",  AVR_ISA_M103,    bfd_mach_avr3},
   {"at43usb320", AVR_ISA_M103,    bfd_mach_avr3},
--- gas/doc/c-avr.texi.old	2007-10-04 20:54:00.531750000 -0600
+++ gas/doc/c-avr.texi	2007-10-04 20:54:55.406750000 -0600
@@ -41,7 +41,7 @@ Instruction set avr2 (default) is for th
 attiny26, at90s2333, at90s2343, at90s4414, at90s4433, at90s4434,
 at90s8515, at90c8534, at90s8535, at86rf401, attiny13, attiny2313,
 attiny261, attiny461, attiny861, attiny24, attiny44, attiny84, attiny25,
-attiny45, attiny85).
+attiny45, attiny85, attiny43u).
 
 Instruction set avr3 is for the classic AVR core with up to 128K program
 memory space (MCU types: atmega103, atmega603, at43usb320, at43usb355,
