Committed
Binutils 2.19
--------------------------------------------------------------------------------
--- gas/config/tc-avr.c.old	2007-10-18 15:59:10.764125000 -0600
+++ gas/config/tc-avr.c	2007-10-19 13:25:41.451886700 -0600
@@ -68,7 +68,6 @@ static struct mcu_type_s mcu_types[] =
   {"avr5",       AVR_ISA_ALL,     bfd_mach_avr5},
   {"avr6",       AVR_ISA_ALL,     bfd_mach_avr6},
   {"at90s1200",  AVR_ISA_1200,    bfd_mach_avr1},
-  {"attiny10",   AVR_ISA_TINY1,   bfd_mach_avr1}, /* XXX -> tn11 */
   {"attiny11",   AVR_ISA_TINY1,   bfd_mach_avr1},
   {"attiny12",   AVR_ISA_TINY1,   bfd_mach_avr1},
   {"attiny15",   AVR_ISA_TINY1,   bfd_mach_avr1},
@@ -99,15 +98,12 @@ static struct mcu_type_s mcu_types[] =
   {"attiny85",   AVR_ISA_TINY2,   bfd_mach_avr2},
   {"attiny43u",  AVR_ISA_TINY2,   bfd_mach_avr2},
   {"attiny48",   AVR_ISA_TINY2,   bfd_mach_avr2},
-  {"atmega603",  AVR_ISA_M603,    bfd_mach_avr3}, /* XXX -> m103 */
   {"atmega103",  AVR_ISA_M103,    bfd_mach_avr3},
   {"at43usb320", AVR_ISA_M103,    bfd_mach_avr3},
   {"at43usb355", AVR_ISA_M603,    bfd_mach_avr3},
   {"at76c711",   AVR_ISA_M603,    bfd_mach_avr3},
   {"atmega48",   AVR_ISA_PWMx,    bfd_mach_avr4},
   {"atmega8",    AVR_ISA_M8,      bfd_mach_avr4},
-  {"atmega83",   AVR_ISA_M8,      bfd_mach_avr4}, /* XXX -> m8535 */
-  {"atmega85",   AVR_ISA_M8,      bfd_mach_avr4}, /* XXX -> m8 */
   {"atmega88",   AVR_ISA_PWMx,    bfd_mach_avr4},
   {"atmega8515", AVR_ISA_M8,      bfd_mach_avr4},
   {"atmega8535", AVR_ISA_M8,      bfd_mach_avr4},
@@ -332,8 +328,8 @@ md_show_usage (FILE *stream)
 	"                   [avr-name] can be:\n"
 	"                   avr1 - AT90S1200, ATtiny1x, ATtiny28\n"
 	"                   avr2 - AT90S2xxx, AT90S4xxx, AT90S8xxx, ATtiny22\n"
-	"                   avr3 - ATmega103, ATmega603\n"
-	"                   avr4 - ATmega83, ATmega85\n"
+	"                   avr3 - ATmega103\n"
+	"                   avr4 - ATmega8, ATmega88\n"
 	"                   avr5 - ATmega161, ATmega163, ATmega32, AT94K\n"
 	"                   or immediate microcontroller name.\n"));
   fprintf (stream,
--- gas/doc/c-avr.texi.old	2007-10-18 16:04:29.936000000 -0600
+++ gas/doc/c-avr.texi	2007-10-18 16:05:48.748500000 -0600
@@ -33,7 +33,7 @@
 Specify ATMEL AVR instruction set or MCU type.
 
 Instruction set avr1 is for the minimal AVR core, not supported by the C
-compiler, only for assembler programs (MCU types: at90s1200, attiny10,
+compiler, only for assembler programs (MCU types: at90s1200,
 attiny11, attiny12, attiny15, attiny28).
 
 Instruction set avr2 (default) is for the classic AVR core with up to
@@ -44,11 +44,11 @@ attiny261, attiny461, attiny861, attiny2
 attiny45, attiny85, attiny43u, attiny48).
 
 Instruction set avr3 is for the classic AVR core with up to 128K program
-memory space (MCU types: atmega103, atmega603, at43usb320, at43usb355,
+memory space (MCU types: atmega103, at43usb320, at43usb355,
 at76c711).
 
 Instruction set avr4 is for the enhanced AVR core with up to 8K program
-memory space (MCU types: atmega48, atmega8, atmega83, atmega85, atmega88,
+memory space (MCU types: atmega48, atmega8, atmega88,
 atmega8515, atmega8535, atmega8hva, at90pwm1, at90pwm2, at90pwm2b,
 at90pwm3, at90pwm3b).
 
