Committed
Binutils 2.19
--------------------------------------------------------------------------------
--- gas/config/tc-avr.c.old	2007-11-07 00:00:58.640625000 -0700
+++ gas/config/tc-avr.c	2007-11-07 00:01:48.500000000 -0700
@@ -100,6 +100,7 @@ static struct mcu_type_s mcu_types[] =
   {"attiny85",   AVR_ISA_TINY2,   bfd_mach_avr2},
   {"attiny43u",  AVR_ISA_TINY2,   bfd_mach_avr2},
   {"attiny48",   AVR_ISA_TINY2,   bfd_mach_avr2},
+  {"attiny88",   AVR_ISA_TINY2,   bfd_mach_avr2},
   {"atmega103",  AVR_ISA_M103,    bfd_mach_avr3},
   {"at43usb320", AVR_ISA_M103,    bfd_mach_avr3},
   {"at43usb355", AVR_ISA_M603,    bfd_mach_avr3},
--- gas/doc/c-avr.texi.old	2007-11-07 00:01:14.781250000 -0700
+++ gas/doc/c-avr.texi	2007-11-07 00:02:18.812500000 -0700
@@ -41,7 +41,7 @@ Instruction set avr2 (default) is for th
 attiny26, at90s2333, at90s2343, at90s4414, at90s4433, at90s4434,
 at90s8515, at90c8534, at90s8535, at86rf401, attiny13, attiny2313,
 attiny261, attiny461, attiny861, attiny24, attiny44, attiny84, attiny25,
-attiny45, attiny85, attiny43u, attiny48).
+attiny45, attiny85, attiny43u, attiny48, attiny88).
 
 Instruction set avr3 is for the classic AVR core with up to 128K program
 memory space (MCU types: atmega103, at43usb320, at43usb355, at76c711).
