Committed
Binutils 2.19
--------------------------------------------------------------------------------
--- include/opcode/avr.h.old	2008-03-25 12:12:41.884348400 -0600
+++ include/opcode/avr.h	2008-03-25 12:16:48.575795000 -0600
@@ -41,6 +41,7 @@
 #define AVR_ISA_RF401 (AVR_ISA_2xxx | AVR_ISA_MOVW | AVR_ISA_LPMX)
 #define AVR_ISA_TINY2 (AVR_ISA_2xxx | AVR_ISA_MOVW | AVR_ISA_LPMX | \
                        AVR_ISA_SPM  | AVR_ISA_BRK)
+#define AVR_ISA_TINY3 (AVR_ISA_TINY2 | AVR_ISA_MEGA)                       
 #define AVR_ISA_M8    (AVR_ISA_2xxx | AVR_ISA_MUL | AVR_ISA_MOVW | \
                        AVR_ISA_LPMX | AVR_ISA_SPM)
 #define AVR_ISA_M603  (AVR_ISA_2xxx | AVR_ISA_MEGA)
--- gas/config/tc-avr.c.old	2008-03-25 12:12:41.884348400 -0600
+++ gas/config/tc-avr.c	2008-03-25 12:17:31.683886600 -0600
@@ -116,6 +116,7 @@ static struct mcu_type_s mcu_types[] =
   {"at76c711",   AVR_ISA_M603,    bfd_mach_avr3},
   {"at90usb82",  AVR_ISA_USB162,  bfd_mach_avr3},
   {"at90usb162", AVR_ISA_USB162,  bfd_mach_avr3},
+  {"attiny167",  AVR_ISA_TINY3,   bfd_mach_avr3},
   {"atmega48",   AVR_ISA_PWMx,    bfd_mach_avr4},
   {"atmega48p",  AVR_ISA_PWMx,    bfd_mach_avr4},
   {"atmega8",    AVR_ISA_M8,      bfd_mach_avr4},
--- gas/doc/c-avr.texi.old	2008-03-25 12:12:41.853088000 -0600
+++ gas/doc/c-avr.texi	2008-03-25 12:17:45.297790800 -0600
@@ -45,7 +45,7 @@ attiny45, attiny85, attiny43u, attiny48,
 
 Instruction set avr3 is for the classic AVR core with up to 128K program
 memory space (MCU types: atmega103, at43usb320, at43usb355, at76c711, 
-at90usb82, at90usb162).
+at90usb82, at90usb162, attiny167).
 
 Instruction set avr4 is for the enhanced AVR core with up to 8K program
 memory space (MCU types: atmega48, atmega48p atmega8, atmega88, atmega88p,
