--- a/include/avr/eeprom.h	2020-05-26 14:04:10.000000000 +0200
+++ b/include/avr/eeprom.h	2020-05-26 11:29:38.153645260 +0200
@@ -112,8 +112,10 @@
  */
 #if	defined (__DOXYGEN__)
 # define eeprom_is_ready()
-#elif	defined (__AVR_XMEGA__) && __AVR_XMEGA__
+#elif	defined (NVM_STATUS)
 # define eeprom_is_ready()	bit_is_clear (NVM_STATUS, NVM_NVMBUSY_bp)
+#elif	defined (NVMCTRL_STATUS)
+# define eeprom_is_ready()	bit_is_clear (NVMCTRL_STATUS, NVMCTRL_EEBUSY_bp)
 #elif	defined (DEECR)
 # define eeprom_is_ready()	bit_is_clear (DEECR, BSY)
 #elif	defined (EEPE)
