diff -Naur binutils/size.c binutils/size.c
--- binutils/size.c	2010-12-13 12:31:10.000000000 -0600
+++ binutils/size.c	2010-12-13 12:38:19.000000000 -0600
@@ -121,6 +121,7 @@
 {
 	{"atxmega256a3",  AVR264K, AVR16K, AVR4K},
 	{"atxmega256a3b", AVR264K, AVR16K, AVR4K},
+	{"atxmega256a3bu",AVR264K, AVR16K, AVR4K},
 	{"atxmega256d3",  AVR264K, AVR16K, AVR4K},
 
 	{"atmega2560",    AVR256K, AVR8K,  AVR4K},
diff -Naur gas/config/tc-avr.c gas/config/tc-avr.c
--- gas/config/tc-avr.c	2010-12-13 12:31:10.000000000 -0600
+++ gas/config/tc-avr.c	2010-12-13 13:14:34.000000000 -0600
@@ -267,6 +267,7 @@
   {"atxmega192d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega256a3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega256a3b",AVR_ISA_XMEGA, bfd_mach_avrxmega6},
+  {"atxmega256a3bu",AVR_ISA_XMEGA,bfd_mach_avrxmega6},
   {"atxmega256d3", AVR_ISA_XMEGA, bfd_mach_avrxmega6},
   {"atxmega128a1", AVR_ISA_XMEGA, bfd_mach_avrxmega7},
   {"atxmega128a1u",AVR_ISA_XMEGA, bfd_mach_avrxmega7},
diff -Naur gas/doc/c-avr.texi gas/doc/c-avr.texi
--- gas/doc/c-avr.texi	2010-12-13 12:31:10.000000000 -0600
+++ gas/doc/c-avr.texi	2010-12-13 13:15:34.000000000 -0600
@@ -103,7 +103,7 @@
 Instruction set avrxmega6 is for the XMEGA AVR core with up to 256K program
 memory space and less than 64K data space (MCU types: atxmega128a3,
 atxmega128d3, atxmega192a3, atxmega128b1, atxmega192d3, atxmega256a3, 
-atxmega256a3b, atxmega192d3).
+atxmega256a3b, atxmega256a3bu, atxmega192d3).
 
 Instruction set avrxmega7 is for the XMEGA AVR core with up to 256K program
 memory space and greater than 64K data space (MCU types: atxmega128a1,
